api/v1alpha1/environment_conversion.go

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/redhat-appstudio/application-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this Memcached to the Hub version (vbeta1).
+func (src *Environment) ConvertTo(dstRaw conversion.Hub) error {
+
```

Comment:
We are missing conversion of the .status field from the old/new version, which causes some of the managed-gitops E2E tests to fail.

See this branch for something that worked for me:
https://github.com/redhat-appstudio/application-api/compare/main...jgwest:application-api:new-env-api?expand=1